kitsu-mal-rating/kitsu-mal-rating.user.js

```diff
@@ -122,41 +130,69 @@
           //Util.log('MAL users rated:', usersRated);
           //Util.log('MAL users faved:', usersFaved);
 
-          var malRatingBar = document.createElement('div');
-          malRatingBar.classList.add('rating-bar');
-          malRatingBar.classList.add('clearfix');
-
-          var ratingElem = document.createElement('span');
-          ratingElem.classList.add('community-percentage');
-          //ratingElem.classList.add('percent-quarter-3');
-          ratingElem.textContent = rating;
-          malRatingBar.appendChild(ratingElem);
-
-          var labelElem = document.createElement('span');
-          labelElem.classList.add('average-rating-stars');
-          labelElem.setAttribute('style', 'top: 5px;');
-          malRatingBar.appendChild(labelElem);
-          var labelLink = document.createElement('a');
-          labelLink.href = malLink;
-          labelLink.setAttribute('target', '_blank');
-          labelElem.appendChild(labelLink);
-          var labelText = document.createElement('h5');
-          labelText.textContent = 'MAL';
-          labelLink.appendChild(labelText);
-
-          var usersElem = document.createElement('span');
-          usersElem.classList.add('ratings-count');
-          usersElem.textContent = usersRated + ' ratings - ' + usersFaved + ' favorites';
-          malRatingBar.appendChild(usersElem);
-
-          waitForElems({
-            sel: '.rating-bar',
-            stop: true,
-            onmatch: function(ratingBar) {
+          var ratingBar = Util.q('.col-sm-8 > section:first-child > div > .rating-bar:first-child:not(#mal-rating-bar)');
+
+          if (barCheck) {
+            var updateRating = Util.q('.community-percentage', barCheck);
+            updateRating.textContent = rating;
+
+            var updateLink = Util.q('a', barCheck);
+            updateLink.href = malLink;
+
+            var updateUsers = Util.q('.ratings-count', barCheck);
+            updateUsers.textContent = usersRated + ' ratings - ' + usersFaved + ' favorites';
+
+            if (!ratingBar.hasAttribute('style')) {
```

Comment:
this line is where the error happens, content gets swapped out between `waitForUrl` firing and `getMalLink` and `barCheck` is out of date.